src/core/watcher.ts

```diff
@@ -246,42 +247,47 @@ class Watcher {
      * the set of files being watched. If a watcher already exists, the file is
      * simply added to the set of files being watched by the existing watcher.
      *
-     * @param {string} filePath - The path of the file to watch.
+     * @param {vscode.Uri | string} uri - The uri of the file to watch.
      */
-    add(filePath: string) {
-        const fileName = path.basename(filePath)
-        const folder = path.dirname(filePath)
+    add(uri: vscode.Uri | string) {
+        if (typeof uri === 'string') {
+            uri = vscode.Uri.file(uri)
+        }
+        const fileName = path.basename(uri.fsPath)
+        const folder = path.dirname(uri.fsPath)
         if (!this.watchers[folder]) {
             this.watchers[folder] = {
-                watcher: this.createWatcher(new vscode.RelativePattern(folder, `*${this.fileExt}`)),
+                watcher: this.createWatcher(new vscode.RelativePattern(folder, `**/*${this.fileExt}`)),
```

Comment:
Inconsistent change. Why recursive?

Reply:
Oops, leftover from testing. I've reverted this change again.